github/utils.go

```diff
@@ -16,24 +18,63 @@ import (
 
 // create service client
 func connect(ctx context.Context, d *plugin.QueryData) *github.Client {
+
+	// Load connection from cache, which preserves throttling protection etc
+	cacheKey := "github"
+	if cachedData, ok := d.ConnectionManager.Cache.Get(cacheKey); ok {
+		return cachedData.(*github.Client)
+	}
+
 	token := os.Getenv("GITHUB_TOKEN")
+	baseURL := os.Getenv("GITHUB_BASE_URL")
 
 	// Get connection config for plugin
 	githubConfig := GetConfig(d.Connection)
-	if githubConfig.Token != nil {
-		token = *githubConfig.Token
+	if &githubConfig != nil {
+		if githubConfig.Token != nil {
+			token = *githubConfig.Token
+		}
+		if githubConfig.BaseURL != nil {
+			baseURL = *githubConfig.BaseURL
+		}
 	}
 
 	if token == "" {
-		panic("'token' must be set in the connection configuration. Edit your connection configuration file and then restart Steampipe")
+		panic("token must be configured")
+		//return nil, errors.New("token must be configured")
 	}
 
 	ts := oauth2.StaticTokenSource(
 		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-	client := github.NewClient(tc)
-	return client
+	conn := github.NewClient(tc)
+
+	// If the base URL was provided then set it on the client. Used for
+	// enterprise installs.
+	if baseURL != "" {
+
+		uv3, err := url.Parse(baseURL)
+		if err != nil {
+			panic(fmt.Sprintf("github.base_url is invalid: %s", baseURL))
+		}
+
+		if uv3.String() != "https://api.github.com/" {
+			uv3.Path = uv3.Path + "api/v3/"
```

Comment:
Do we need to include this, or does the go-github library automatically attach this suffix, per https://github.com/google/go-github/blob/a58d5f0adf74ef73aadaac476f7285afc77a3b43/github/github.go#L357-L364